src/components/templates/ReactionPage/ReactionList.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useMemo, useState } from "react";
+import classNames from "classnames";
+
+import { DEFAULT_PARTY_NAME, DEFAULT_PROFILE_IMAGE } from "settings";
+
+import {
+  chatMessageAsMessageToTheBand,
+  Reaction,
+  ReactionsTextMap,
+} from "utils/reactions";
+import { withId, WithId } from "utils/id";
+
+import { User } from "types/User";
+import { ChatMessage } from "types/chat";
+
+import { useWorldUsersByIdWorkaround } from "hooks/users";
+
+import UserProfileModal from "components/organisms/UserProfileModal";
+import UserProfilePicture from "components/molecules/UserProfilePicture";
+import { UserAvatar } from "components/atoms/UserAvatar";
+
+export interface ReactionListProps {
+  reactions: Reaction[];
+  chats: ChatMessage[];
+  small?: boolean;
+}
+
+export const ReactionList: React.FC<ReactionListProps> = ({
+  reactions,
+  chats,
+  small = false,
+}) => {
+  // @debt see comments in useWorldUsersByIdWorkaround
+  const { worldUsersById } = useWorldUsersByIdWorkaround();
+
+  const [selectedUserProfile, setSelectedUserProfile] = useState<
+    WithId<User>
+  >();
+
+  const allReactions = useMemo(() => {
+    const chatsAsBandMessages = chats?.map(chatMessageAsMessageToTheBand) ?? [];
+
+    const allReactionsSorted = [
+      ...(reactions ?? []),
+      ...chatsAsBandMessages,
+    ].sort((a, b) => b.created_at - a.created_at);
```

Comment:
could we have `chatSort` here?

Reply:
I did think about that, but then didn’t go and look if it would work for this. I believe these are technically ‘messages to the band’ format, so it may not work for it

Reply:
I guess since it uses the same `Message` type, it should

Reply:
It doesn't though, which is what I meant by the following above:

> I believe these are technically ‘messages to the band’ format,

![image](https://user-images.githubusercontent.com/753891/109251058-ed54df80-783e-11eb-85a8-95f5d986ed08.png)

![image](https://user-images.githubusercontent.com/753891/109251077-f5148400-783e-11eb-894a-e8b2148ec72b.png)

So the types are `Reaction` and `MessageToTheBandReaction`, which since `Reaction` is the generic 'grouping' type, will be what they both become.

This is defined in `src/utils/reactions.ts`:

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/reactions.ts#L86-L99

and seems to use `created_at`, which is a `number`.

`chatSort` works on `ts_utc`, which is a `firebase.firestore.Timestamp`:

- https://github.com/sparkletown/sparkle/blob/staging/src/utils/chat.ts#L15-L18

---

@mike-lvov So, roundabout path to a researched answer, no we can't use `chatSort` here.